### ISSUE TITLE
MachineDeployment sometimes doubles EC2 instances on AWS

### ISSUE THREAD

**Issue by kirek007** [kubermatic/kubermatic#6960]
**User Story**
From time to time crating node pools creates doubled EC2 instances. 
Only requested amount of nodes will join other just hang in AWS console as deployed and healthy.
Names of there doubles machines are also doubled:

AWS EC2 list:
```
  | worker-two-7d9d674596-hfmzt | i-088667b5e346bb9db | Running | 
  | worker-two-7d9d674596-p26ns | i-04f1fa24d1b4e10c9 | Running |
  | worker-two-7d9d674596-hfmzt | i-0745f6bff9e4659ea | Running |
  | worker-two-7d9d674596-p26ns | i-0dadcba0c0ea71cbb | Running |
```

Kubermatic:
Nodes list:
![image](https://user-images.githubusercontent.com/1878027/116387443-1c75d600-a81b-11eb-8ca8-1804c54fffb4.png)
Events:
![image](https://user-images.githubusercontent.com/1878027/116387541-33b4c380-a81b-11eb-8f4b-e25bff76b5ba.png)

As you can see, event `Successfully created instance` is doubled and both events has count equals 2.

**Acceptance criteria**
I want to have only as many EC2 instances as requested nodes in MachineDeployment.

**Comment by kirek007** [comment]
Looks like there additional EC2 sometimes joins cluster, so it's not consistent between UI and real state.

**Machines that should be created**
```bash
$ k get machinedeployments,machines,machineset -A
NAMESPACE     NAME                                               AGE   DELETED   REPLICAS   AVAILABLEREPLICAS   PROVIDER   OS        VERSION
kube-system   machinedeployment.cluster.k8s.io/worker-flat-car   11m             2          1                   aws        flatcar   1.19.3
kube-system   machinedeployment.cluster.k8s.io/worker-one        35m             3          3                   aws        ubuntu    1.19.3
kube-system   machinedeployment.cluster.k8s.io/worker-two        27m             2          2                   aws        ubuntu    1.19.3

NAMESPACE     NAME                                                     AGE   DELETED   MACHINESET                  ADDRESS        NODE                                         PROVIDER   OS        VERSION
kube-system   machine.cluster.k8s.io/worker-flat-car-56d4478b8-2l7ww   11m             worker-flat-car-56d4478b8   172.16.3.242   ip-172-16-3-242.us-west-2.compute.internal   aws        flatcar   1.19.3
kube-system   machine.cluster.k8s.io/worker-flat-car-56d4478b8-wrj8c   11m             worker-flat-car-56d4478b8   172.16.3.51    ip-172-16-3-51.us-west-2.compute.internal    aws        flatcar   1.19.3
kube-system   machine.cluster.k8s.io/worker-one-7c65d5cb77-4pj6j       35m             worker-one-7c65d5cb77       172.16.1.224   ip-172-16-1-224.us-west-2.compute.internal   aws        ubuntu    1.19.3
kube-system   machine.cluster.k8s.io/worker-one-7c65d5cb77-68pfl       35m             worker-one-7c65d5cb77       172.16.1.123   ip-172-16-1-123.us-west-2.compute.internal   aws        ubuntu    1.19.3
kube-system   machine.cluster.k8s.io/worker-one-7c65d5cb77-pdtwk       35m             worker-one-7c65d5cb77       172.16.1.152   ip-172-16-1-152.us-west-2.compute.internal   aws        ubuntu    1.19.3
kube-system   machine.cluster.k8s.io/worker-two-7d9d674596-hfmzt       27m             worker-two-7d9d674596       172.16.2.25    ip-172-16-2-25.us-west-2.compute.internal    aws        ubuntu    1.19.3
kube-system   machine.cluster.k8s.io/worker-two-7d9d674596-p26ns       27m             worker-two-7d9d674596       172.16.2.7     ip-172-16-2-7.us-west-2.compute.internal     aws        ubuntu    1.19.3

NAMESPACE     NAME                                                  AGE   DELETED   REPLICAS   AVAILABLEREPLICAS   MACHINEDEPLOYMENT   PROVIDER   OS        VERSION
kube-system   machineset.cluster.k8s.io/worker-flat-car-56d4478b8   11m             2          1                   worker-flat-car     aws        flatcar   1.19.3
kube-system   machineset.cluster.k8s.io/worker-one-7c65d5cb77       35m             3          3                   worker-one          aws        ubuntu    1.19.3
kube-system   machineset.cluster.k8s.io/worker-two-7d9d674596       27m             2          2                   worker-two          aws        ubuntu    1.19.3
```

**List of nodes that joined cluster:** (look at IPs)
```bash
$ k get no -o wide
NAME                                         STATUS   ROLES    AGE     VERSION   INTERNAL-IP    EXTERNAL-IP   OS-IMAGE                                                 KERNEL-VERSION     CONTAINER-RUNTIME
ip-172-16-1-123.us-west-2.compute.internal   Ready    <none>   31m     v1.19.3   172.16.1.123   <none>        Ubuntu 18.04.5 LTS                                       5.4.0-1045-aws     docker://19.3.12
ip-172-16-1-152.us-west-2.compute.internal   Ready    <none>   32m     v1.19.3   172.16.1.152   <none>        Ubuntu 18.04.5 LTS                                       5.4.0-1045-aws     docker://19.3.12
ip-172-16-1-224.us-west-2.compute.internal   Ready    <none>   30m     v1.19.3   172.16.1.224   <none>        Ubuntu 18.04.5 LTS                                       5.4.0-1045-aws     docker://19.3.12
ip-172-16-2-25.us-west-2.compute.internal    Ready    <none>   24m     v1.19.3   172.16.2.25    <none>        Ubuntu 18.04.5 LTS                                       5.4.0-1045-aws     docker://19.3.12
ip-172-16-2-65.us-west-2.compute.internal    Ready    <none>   24m     v1.19.3   172.16.2.65    <none>        Ubuntu 18.04.5 LTS                                       5.4.0-1045-aws     docker://19.3.12
ip-172-16-2-7.us-west-2.compute.internal     Ready    <none>   24m     v1.19.3   172.16.2.7     <none>        Ubuntu 18.04.5 LTS                                       5.4.0-1045-aws     docker://19.3.12
ip-172-16-2-92.us-west-2.compute.internal    Ready    <none>   23m     v1.19.3   172.16.2.92    <none>        Ubuntu 18.04.5 LTS                                       5.4.0-1045-aws     docker://19.3.12
ip-172-16-3-192.us-west-2.compute.internal   Ready    <none>   8m36s   v1.19.3   172.16.3.192   <none>        Flatcar Container Linux by Kinvolk 2345.3.1 (Rhyolite)   4.19.107-flatcar   docker://18.6.3
ip-172-16-3-242.us-west-2.compute.internal   Ready    <none>   8m32s   v1.19.3   172.16.3.242   <none>        Flatcar Container Linux by Kinvolk 2345.3.1 (Rhyolite)   4.19.107-flatcar   docker://18.6.3
ip-172-16-3-25.us-west-2.compute.internal    Ready    <none>   8m32s   v1.19.3   172.16.3.25    <none>        Flatcar Container Linux by Kinvolk 2345.3.1 (Rhyolite)   4.19.107-flatcar   docker://18.6.3
ip-172-16-3-51.us-west-2.compute.internal    Ready    <none>   7m31s   v1.19.3   172.16.3.51    <none>        Flatcar Container Linux by Kinvolk 2345.3.1 (Rhyolite)   4.19.107-flatcar   docker://18.6.3
```

**Comment by kron4eg** [comment]
This was fixed in: https://github.com/kubermatic/machine-controller/pull/886 and back ported to machine-controller 1.24.4 https://github.com/kubermatic/machine-controller/pull/887 however seems like we forgot to update kubermatic itself. So #6962 will address exactly this. Once done, we will release new v2.16.9 I suppose.

**Comment by kirek007** [comment]
Great, I'll test it as soon as release will be available.

**Comment by kron4eg** [comment]
Updated machine-controller with fix is released as part of the v2.16.9 and v2.15.12 kubermatic releases.
